components/core/buttons/TextButton.vue

```diff
@@ -73,25 +99,28 @@ export default {
     outline: none;
     font-weight: 700;
     font-size: 1rem;
-    background-color: transparent;
 }
 
-.core-button.--rounded {
+.core-button:not(.--is-link):hover,
+.core-button.--is-link > a:hover {
+    color: #7568f6;
+    border-color: #7568f6;
+}
+
+.--rounded:not(.--is-link),
+.--rounded.--is-link > a {
     border-radius: 9999px;
 }
 
-.core-button.--primary {
+.--rounded:not(.--is-link),
+.--rounded.--is-link > a {
     color: #c2a53a;
     border: 0.25rem solid #c2a53a;
 }
 
-.core-button.--secondary {
+.--secondary:not(.--is-link),
+.--secondary.--is-link > a {
```

Comment:
The CSS selector works fine. But let's still declare the `.core-button` altogether:

```suggestion
.core-button.--secondary:not(.--is-link),
.core-button.--secondary.--is-link > a {
```

Please also apply the same selector to the other variants - `.--rounded`, `.--primary`

Reply:
Okay, I got it.
The reason for declaring `.core-button` is to make it more specific right?

Reply:
Yes, exactly. Since you mentioned, you could also look into “specificity” in CSS.
Besides, this also shows the consistency of using the same parent.